internal/collectors/builder.go

```diff
@@ -0,0 +1,448 @@
+/*
+Copyright 2018 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package collectors
```

Comment:
As far as I know, golang packages should be singular. Would that be ok for you?

Reply:
I am fine with `collector`, but maybe we could come up with a more clear name, as it's a collection of object collectors. Might be a bit confusing with the already `pkg/collector`?

Reply:
I would even say the similarity helps users understand the link between the two. Not a strong opinion.